source/setup.Rmd

```diff
@@ -1,282 +1,277 @@
+sudo chmod u+x get-docker.sh
+sudo sh get-docker.sh
 ```
 
-**Ubuntu:** To install Git on Ubuntu, open the terminal 
-and type the following commands:
-
+**Running JupyterLab** In the terminal, navigate to the directory where you want to run JupyterLab, and run
+the following command.
 ```
-sudo apt update
-sudo apt install git
+docker run --rm -v $(pwd):/home/jovyan/work -p 8888:8888 ubcdsci/r-dsci-100:20230724142640d8f06f jupyter lab
 ```
+The terminal will then print some text as the Docker container starts. Once the text stops scrolling, find the 
+URL in your terminal that starts with `http://127.0.0.1:8888` (highlighted by the red box Figure \@ref(fig:ubuntu-docker-terminal)), and paste it into your browser to start JupyterLab.
 
+```{r ubuntu-docker-terminal, echo = FALSE, fig.cap = "The terminal text after running the Docker container in Ubuntu. The red box indicates the URL that you should paste into your browser to open JupyterLab.", fig.retina = 2, out.width="100%"}
+image_read("img/setup/ubuntu-docker.png") |> 
+  image_crop("3632x2000")
+```
 
-### Miniconda
-
-To run Jupyter notebooks on your computer, 
-you will need to install the web-based platform JupyterLab. 
-But JupyterLab relies on Python, so we need to install Python first.
-We can install Python via
-the \index{miniconda} [miniconda Python package distribution](https://docs.conda.io/en/latest/miniconda.html).
-
-**Windows:** To install miniconda on Windows, download 
-the [latest Python 64-bit version from here](https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe). 
-Once the download has finished, run the installer 
-and accept the default configuration for all pages. 
-After installation, you can open the Anaconda Prompt
-by opening the Start Menu and searching for the program called 
-"Anaconda Prompt (miniconda3)". 
-When this opens, you will see a prompt similar to 
-`(base) C:\Users\your_name`. 
+More information on installing and running Docker on Ubuntu, as well as troubleshooting tips, can be found
+[in the online Docker documentation](https://docs.docker.com/engine/install/ubuntu/).
+
+### MacOS
+
+**Installation** To install Docker on MacOS,
+visit [the online Docker documentation](https://docs.docker.com/desktop/install/mac-install/), and
+download the `Docker.dmg` installation file that is appropriate for your
+computer. To know which installer is right for your machine, you need to know 
+whether your computer has an Intel processor (older machines) or an
+Apple processor (newer machines); the [Apple support page](https://support.apple.com/en-ca/HT211814) has
+information to help you determine which processor you have. Once downloaded, double-click 
+the file to open the installer, then drag the Docker icon to the Applications folder.
+Double-click the icon in the Applications folder to start Docker. In the installation 
+window, use the recommended settings.
+
+**Running JupyterLab** Once Docker is installed and running, follow the
```

Comment:
This section also leads me to believe we should have the Windows section first, given that we are telling students to scroll down to that section and follow the instructions there?

Reply:
yup, sounds fine to me